packages/eslint-plugin/tests/rules/naming-convention/naming-convention.test.ts

```diff
@@ -13,6 +16,38 @@ const parserOptions = {
   project: './tsconfig.json',
 };
 
+/** Union of all case error configs with `messageId` as discriminant property for type narrowing */
+type CaseErrorWithDataConfig =
+  | {
+      messageId: 'doesNotMatchFormat';
+      type: Capitalize<SelectorsString>;
+      name: string;
+      expectedFormats: PredefinedFormatsString[];
+    }
+  | { messageId: 'unexpectedUnderscore' }; // todo add more detail and union members as required
```

Comment:
Sorry, I don't follow this todo. What's the intent? Is it blocking the PR being merged?

Reply:
Nope, its not blocking the PR being merged, I had made this so its clearer what options are available when asserting errors for different messages, I had defined the ones I was concerned with for this PR but there are other messages not in the union, so left the todo as an indicator that the list wasn't complete.

Reply:
but based on your other comment will remove this typing